### DISCUSSION TITLE
Feature ideas

### DISCUSSION THREAD

**Discussion by bkw777** [bkw777/mainline#225] [General]
* A single button that does both install-latest and uninstall-old in one shot, and with a single admin password prompt.

And a config option to go with that to make the notification button do that , or just install.

You would always actually have a rolling 2 kernels installed, because uninstall-old will always leave behind the one you're currently running, and it's not worth setting up some contraption to schedule the uninstall to do itself after reboot. The leftover kernel will get uninstalled at the next uninstall-old. You'd always have one extra kernel installed, but they won't accumulate, it would just always be the latest and one other.

* a way to queue up a bunch of different install and remove ops and fire off a single script to do them all with a single admin password prompt.

So instead of hitting Install on a single kernel, you select a bunch, and they all get either installed or uninstalled, whichever they aren't to begin with. I promis not to label the button XOR, but I really want to. ;)

I almost never install more than one at at time though so maybe there is no need. But it seems natural and good even if not all that often used.

There are many other wish list things too of course like accessing the non-generic builds. Discuss whatever.

**Comment by AtomicRobotMan0101** [comment]
As a comment only, Mint by default has the current and two trailing.

I do not know why. Safety? Super precautionary for noobs?

I like the idea of auto-removal.

**Comment by bkw777** [comment]
It must be to cover the case where you leave a machine running long enough to receive more than one new kernel update, only to discover that the new kernel doesn't boot, or has no working video, or has no working network, and so you need to go back, but the immediate previous kernel is also broken, but you never noticed since you never booted into that one. So they keep yet another one. I don't know how common that case is. Maybe if Mint happens to push out kernel updates frequently that could actually happen easily.

**Comment by nbohr1more** [comment]
In Linux Mint 21.1 if you have Secure Boot enabled the Nvidia driver installer will offer you the option to sign the Nvidia Kernel Module.
It might be nice to integrate that feature into the mainline installer so that users wont need to externally sign their kernels?
